projects/packages/sync/src/modules/class-options.php

```diff
@@ -364,9 +364,13 @@ private function filter_theme_mods( &$value ) {
 	public function jetpack_sync_core_icon() {
 		$url = get_site_icon_url();
 
-		require_once JETPACK__PLUGIN_DIR . 'modules/site-icon/site-icon-functions.php';
+		$jetpack_url = \Jetpack_Options::get_option( 'site_icon_url' );
+		if ( function_exists( 'jetpack_site_icon_url' ) || ( defined( 'JETPACK__PLUGIN_DIR' ) && include_once JETPACK__PLUGIN_DIR . 'modules/site-icon/site-icon-functions.php' ) ) {
```

Comment:
Oof, same here, this is too long and not very readable, can we fix it please?

Reply:
Agreed, this pattern was bothering me a bit too.
Fixed it.